File: students/k3340/Kuznetsov_Artur/Lr1/src/task1/UDPClient.java
package task1;

import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;

public class UDPClient {

    // UDP не требует установки соединения. Клиент просто отправляет датаграмму (пакет данных),
    // а сервер принимает её, не устанавливая явного соединения
    public static void main(String[] args) {
        int serverPort = 8000;
        int bufferSize = 1024;
        String message = "Hello, server!";

        try {
            // Сокет для отправки и получения данных
            DatagramSocket clientSocket = new DatagramSocket();

            // Отправка сообщения серверу
            InetAddress serverAddress = InetAddress.getByName("localhost");
            byte[] sendData = message.getBytes();
            DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length, serverAddress, serverPort);
            clientSocket.send(sendPacket);

            // Прием ответа от сервера
            byte[] receiveData = new byte[bufferSize];
            DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);
            clientSocket.receive(receivePacket);

            // Вывод ответа сервера
            String response = new String(receivePacket.getData()).trim();
            System.out.println("Ответ от сервера: " + response);

            clientSocket.close();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: students/k3341/Budunov_Budun/laboratory_works/laboratory_work_1/Task_1/udp_client.py
import socket


def start_client():
    # Создаем UDP-сокет
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

    # Адрес сервера
    server_address = ('localhost', 8080)

    # Отправляем сообщение серверу
    message = "Hello, server"
    client_socket.sendto(message.encode(), server_address)
    print(f"Отправлено сообщение серверу: {message}")

    # Получаем ответ от сервера
    response, _ = client_socket.recvfrom(1024)
    print(f"Получено сообщение от сервера: {response.decode()}")

    # Закрываем сокет
    client_socket.close()


if __name__ == "__main__":
    start_client()
